### DISCUSSION TITLE
Feature Request: Support for Loras in Wildcards

### DISCUSSION THREAD

**Discussion by dhosein** [lllyasviel/Fooocus#3193] [Ideas]
I know that this was looked at with the recent update that now allows Loras to be included in prompts, and that the solution was non-trivial. But being able to include Loras in wildcards, for me, would be the single biggest improvement to Fooocus that I could really make use of. At least, biggest short of switching the model to SD3, or the LLM to GPT-3 or GPT-4

I use wildcard to introduce random selections of all sorts... characters, visual styles, poses etc...  and there are a lot of Loras out there that enhance these, but I can't really make use of them at present since I don't know which lora I want to include.. the ones I want are determined by the random outcome with way more than 5 possibilities (So I can't simply list them all).

**Comment by mashb1t** [comment]
Relates to https://github.com/lllyasviel/Fooocus/pull/2966 and is currently not possible without substantially changing the structure, which would lead to multiple times having to refresh the model, which can be (very) slow depending on the system.
You can change this yourself when reloading all pipeline steps after refresh_everything by duplicating large portions of the code or completely restructuring it. I already did significantly change the code of the async_worker when implementing https://github.com/mashb1t/Fooocus/discussions/42 (diff [here](https://github.com/mashb1t/Fooocus/compare/v2.4.3...mashb1t:Fooocus:feature/multi-step-rendering#diff-64c538b31e5ff03a06701a56b8033cdcf31b656d50075e9d2f25f30e3ca6ed6a)), but this still doesn't fully allow for wildcards to use LoRAs.
Please check this out nevertheless.
